store/postgres/src/entities.rs

```diff
@@ -0,0 +1,993 @@
+            // that we call it Integer here
+            event_id -> Integer,
+            entity_id -> Varchar,
+            subgraph -> Varchar,
+            entity -> Varchar,
+            data_before -> Nullable<Jsonb>,
+            data_after -> Nullable<Jsonb>,
+            op_id -> SmallInt,
+            reversion -> Bool,
+        }
+    }
+
+    table! {
+        event_meta_data (id) {
+            id -> Integer,
+            db_transaction_id -> BigInt,
+            db_transaction_time -> Timestamp,
+            source -> Nullable<Varchar>,
+        }
+    }
+
+    joinable!(entity_history -> event_meta_data (event_id));
+    allow_tables_to_appear_in_same_query!(entity_history, event_meta_data);
+
+    table! {
+        sgd_schemas(id) {
+            id -> Integer,
+            subgraph -> Text,
+            name -> Text,
+        }
+    }
+}
+
+// The entities table for the subgraph of subgraphs.
+mod subgraphs {
+    table! {
+        subgraphs.entities (entity, id) {
+            entity -> Varchar,
+            id -> Varchar,
+            data -> Jsonb,
+            event_source -> Varchar,
+        }
+    }
+}
+
+impl EntitySource for self::public::entities::table {}
+
+impl EntitySource for self::subgraphs::entities::table {}
+
+// This is a bit weak, as any DynamicTable<String> is now an EntitySource
+impl EntitySource for DynamicTable<String> {}
+
+/// Support for the management of the schemas and tables we create in
+/// the database for each deployment. The Postgres schemas for each
+/// deployment/subgraph are tracked in the `sgd_schemas` table.
+///
+/// The functions in this module are very low-level and should only be used
+/// directly by the Postgres store, and nowhere else. At the same time, all
+/// manipulation of entities in the database should go through this module
+/// to make it easier to handle future schema changes
+// We use Diesel's dynamic table support for querying the entities and history
+// tables of a subgraph. Unfortunately, this support is not good enough for
+// modifying data, and we fall back to generating literal SQL queries for that.
+// For the `entities` table of the subgraph of subgraphs, we do map the table
+// statically and use it in some cases to bridge the gap between dynamic and
+// static table support, in particular in the update operation for entities.
+// Diesel deeply embeds the assumption that all schema is known at compile time;
+// for example, the column for a dynamic table can not implement
+// `diesel::query_source::Column` since that must carry the column name as a
+// constant. As a consequence, a lot of Diesel functionality is not available
+// for dynamic tables.
+use public::sgd_schemas;
+
+#[derive(Queryable, QueryableByName, Debug)]
+#[table_name = "sgd_schemas"]
+struct Schema {
+    id: i32,
+    subgraph: String,
+    name: String,
+}
+
+type EntityColumn<ST> = Column<DynamicTable<String>, String, ST>;
+
+#[derive(Debug, Clone)]
+pub(crate) struct SplitTable {
```

Comment:
What does split table mean here?

Reply:
IMHO splitting is just the action we're taking here, I'd give this a more permanent name. `DeploymentEntitiesTable` or something similar.

Reply:
What confused me here was that I thought the table in itself was split somehow.

Reply:
Yeah, I should have explained in a comment more clearly what that's for; added a bit of an explanation

Reply:
I still find `Split` and `SplitTable` to not be great names. `Dedicated`? `PerSubgraph`, `Subgraph`, `DeploymentEntities` (putting it back on the table 😋)...?